### PR TITLE
Ajax submission of a single answer for both worksheet and guided

### LINKED ISSUE #1167: Use Ajax to submit answers for evaluation

### PR DESCRIPTION
**Pull request by allenwq**
To replace #1181, Fix #1167 and #1168 

Change the redirection of jobs to ajax polling jobs.
Support submit programming questions in worksheet submission.

### CODE REVIEW

**Comment by allenwq** [comment]
I will fix the coverage in a later PR when implement auto reattempt.

File: app/assets/javascripts/course/assessment/submission/answer/programming_submit.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    if (data.status == 'completed') {
+      reload_answer(answer_id);
+    } else if (data.status == 'submitted') {
+      waitForJob(url, answer_id);
+    } else {
+      // Show errors.
+    }
+  }
+
+  function reload_answer(answer_id) {
```
Comment (kxmbrian) [inline app/assets/javascripts/course/assessment/submission/answer/programming_submit.js]: function name should use camel case not underscore

Comment (allenwq) [inline app/assets/javascripts/course/assessment/submission/answer/programming_submit.js]: Sorry, I guess I code ruby too much.

**Comment by kxmbrian** [comment]
Since `Course::Assessment::Submission::UpdateWorksheetAssessmentService#update` and `Course::Assessment::Submission::UpdateWorksheetAssessmentService#update` are identical, is it better to move it up and merge it with `Course::Assessment::Submission::UpdateService#update`?

**Comment by allenwq** [comment]
Good catch ! Actually all the attempting code will be removed in my next PR, but let me merge them here first.

**Pull request merged**